src/mawpy/steps/incremental_clustering.py

```diff
@@ -201,45 +113,43 @@ def _k_means_cluster_lloyd(cluster_list: list[Cluster]) -> list[Cluster]:
             Cluster()
```

Comment:
I suggest adding a `list[points]` parameter to the Cluster class so that you can add all the points in one operation and recompute the center only once.

Reply:
added the function.